safe_transaction_service/history/indexers/internal_tx_indexer.py

```diff
@@ -367,7 +339,18 @@ def process_elements(
             logger.debug(
                 "End decoding and storing of %d decoded traces", internal_txs_decoded
             )
-            return tx_hashes
+
+        # Mark traces as processed
+        for tx_hash in list(tx_hash_with_traces.keys()):
+            block_hash = (
+                tx_hash_with_traces[tx_hash][0]["blockHash"]
+                if tx_hash_with_traces[tx_hash]
+                else None
+            )
+            self.element_already_processed_checker.mark_as_processed(
+                tx_hash, block_hash
```

Comment:
Could be the same transaction with different trace_address to be a problem here?

Reply:
No, as all the traces are processed together for a `tx_hash`